### ISSUE TITLE
No notification because of SOFT service recovery 

### ISSUE THREAD

**Issue by orbison** [NagiosEnterprises/nagioscore#788]
Hi there,

please check the following sequence of events:

> [Sun Sep 13 21:15:40 2020] HOST ALERT: sw-hosting-1;**DOWN;SOFT**;1;PING CRITICAL - Packet loss = 100%
> [Sun Sep 13 21:15:55 2020] HOST ALERT: sw-hosting-1;UNREACHABLE;SOFT;2;PING CRITICAL - Packet loss = 100%
> [Sun Sep 13 21:16:47 2020] HOST ALERT: sw-hosting-1;UNREACHABLE;SOFT;3;PING CRITICAL - Packet loss = 100%
> [Sun Sep 13 21:17:05 2020] SERVICE ALERT: sw-hosting-1;Port-channel10 - INTERNET;**WARNING;HARD**;1;CLUSTER WARNING: Port-channel10 - INTERNET: 1 ok, 0 warning, 1 unknown, 0 critical
> [Sun Sep 13 21:17:47 2020] HOST ALERT: sw-hosting-1;UNREACHABLE;SOFT;4;PING CRITICAL - Packet loss = 100%
> [Sun Sep 13 21:18:48 2020] HOST ALERT: sw-hosting-1;**UNREACHABLE;HARD**;5;CRITICAL - Host Unreachable (10.29.0.2)
> [Sun Sep 13 21:21:07 2020] HOST ALERT: sw-hosting-1;**UP;HARD**;1;PING OK - Packet loss = 0%, RTA = 3.17 ms
> [Sun Sep 13 21:26:05 2020] SERVICE NOTIFICATION: ppokorny;sw-hosting-1;Port-channel10 - INTERNET;WARNING;service-notify-by-neno;CLUSTER WARNING: Port-channel10 - INTERNET: 1 ok, 0 warning, 1 unknown, 0 critical
> [Sun Sep 13 21:26:05 2020] SERVICE NOTIFICATION: pzvonik;sw-hosting-1;Port-channel10 - INTERNET;WARNING;service-notify-by-neno;CLUSTER WARNING: Port-channel10 - INTERNET: 1 ok, 0 warning, 1 unknown, 0 critical
> [Sun Sep 13 21:26:05 2020] SERVICE NOTIFICATION: ppokorny-sms;sw-hosting-1;Port-channel10 - INTERNET;WARNING;service-notify-by-sms;CLUSTER WARNING: Port-channel10 - INTERNET: 1 ok, 0 warning, 1 unknown, 0 critical
> [Sun Sep 13 21:26:05 2020] SERVICE NOTIFICATION: pzvonik-sms;sw-hosting-1;Port-channel10 - INTERNET;WARNING;service-notify-by-sms;CLUSTER WARNING: Port-channel10 - INTERNET: 1 ok, 0 warning, 1 unknown, 0 critical
> [Sun Sep 13 21:29:05 2020] SERVICE ALERT: sw-hosting-1;Port-channel10 - INTERNET;**OK;SOFT**;1;CLUSTER OK: Port-channel10 - INTERNET: 2 ok, 0 warning, 0 unknown, 0 critical

The customer got never notified on the recovery of the Port-channel10 service and that seems like a bug to me.

The behaviour is similar to that in [bug #759](https://github.com/NagiosEnterprises/nagioscore/issues/759), but it is not exactly the same — the HOST recovery was HARD in our case.

I am not sure if the behaviour we have observed is correct or not, but from the customer's point of view, it is not desirable: they were notified on the Port-channel10 going down, but they never got a notification about it going up again.

And on the same note: we cannot figure out why the notification was sent nine minutes after the service entered a HARD WARNING state. No notification delay has been configured on that particular service. The service definition is as follows (extracted from objects.cache):

```
define service {
        host_name       sw-hosting-1
        service_description     Port-channel10 - INTERNET
        check_period    24x7
        check_command   check-service-cluster!Port-channel10 - INTERNET!@1:!@2:!$SERVICESTATEID:sw-hosting-1:Gi1/0/3 status /internet - part of Po10/$,$SERVICESTATEID:sw-hosting-1:Gi2/0/3 status /internet - part of Po10/$
        contacts        pzvonik-sms,ppokorny-sms,pzvonik,ppokorny
        notification_period     24x7
        initial_state   o
        importance      0
        check_interval  3.000000
        retry_interval  1.000000
        max_check_attempts      4
        is_volatile     0
        parallelize_check       1
        active_checks_enabled   1
        passive_checks_enabled  1
        obsess  0
        event_handler_enabled   1
        low_flap_threshold      0.000000
        high_flap_threshold     0.000000
        flap_detection_enabled  1
        flap_detection_options  a
        freshness_threshold     0
        check_freshness 0
        notification_options    r,w,u,c,f
        notifications_enabled   1
        notification_interval   0.000000
        first_notification_delay        0.000000
        stalking_options        n
        process_perf_data       1
        retain_status_information       1
        retain_nonstatus_information    1
        _NENO_TEMPLATE  service
        }
```
Why the delay was _exactly_ nine minutes? At 21:17:05 the service went to WARNING HARD state and at 21:26:05 the notification was generated.

Attached please find the complete log file.

Kind regards,

Rostislav
[nagios-09-14-2020-00.log](https://github.com/NagiosEnterprises/nagioscore/files/5226998/nagios-09-14-2020-00.log)
